### DISCUSSION TITLE
Unable to start MinIO Gateway on ECS2 instance

### DISCUSSION THREAD

**Discussion by ahmedriza** [minio/minio#12958] [Q&A] [answered]
This is hopefully something simple that I have overlooked:

We have an EC2 instance with the correct IAM roles that allows us to use S3 (read, write etc).  For example, the following works fine:
```
aws s3 ls s3://some-bucket
```

When we try to start a MinIO Gateway instance here, however, it fails with the following error message:
```
./minio gateway s3

ERROR Unable to initialize gateway backend: Get "https://s3.amazonaws.com/probe-bucket-sign-XXXX/?location=": proxyconnect tcp: x509: cannot validate certificate for XX.XX.XX.XX because it doesn't contain any IP SANs
```
where ` XX.XX.XX.XX` is an HTTP proxy which proxies all http/https traffic on our EC2 instances. 

Any hints/clues as to where we need to look into tracking this issue?  Thanks.

**Comment by vadmeste** [comment, marked as answer]
> ERROR Unable to initialize gateway backend: Get "https://s3.amazonaws.com/probe-bucket-sign-XXXX/?location=": proxyconnect tcp: x509: cannot validate certificate for XX.XX.XX.XX because it doesn't contain any IP SANs

I would say you have a proxy with a certificate that golang doesn't like. If your proxy has a FQDN address stored in the certificate, then you need to use that address. Otherwise you will have to regenerate your proxy certificate properly.

**Reply by ahmedriza** [reply]
@vadmeste thanks for the suggestion.  After changing the Proxy to the FQDN, the certificate error is resolved.  

Got a new error now: 
```
ERROR Unable to initialize gateway backend: Get "https://s3.amazonaws.com/probe-bucket-sign-XXX/?location=": Connection closed by foreign host https://s3.amazonaws.com/probe-bucket-sign-klgt06nmj23p/?location=. Retry again.
```
Wonder if there's a way to get more logging from minio as to what the cause might be.  Could this be our proxy again causing problems?

**Reply by ahmedriza** [reply]
Resolved: had to specify the s3 region when starting the gateway as well ensuring that the NO_PROXY environment variable was set correctly, so that the proxy is not used when connecting to hosts inside AWS.

Thanks for your help.
